clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -269,7 +269,7 @@ public int partitionLeaderEpoch() {
 
     public InputStream recordInputStream(BufferSupplier bufferSupplier) {
         final ByteBuffer buffer = this.buffer.duplicate();
-        buffer.position(RECORDS_OFFSET);
+        buffer.position(buffer.position() + RECORDS_OFFSET);
```

Comment:
not a relates change but something I found while lurking in the code. I would probably revert it from this pr so that we don't pollute this with unnecessary changes.